packages/expo-updates/android/src/main/java/expo/modules/updates/manifest/LegacyUpdateManifest.kt

```diff
@@ -114,7 +114,7 @@ class LegacyUpdateManifest private constructor(
             Date()
           }
         } else {
-          Date()
+          throw JSONException("Missing commitTime")
```

Comment:
`commitTime` isn't a required field in legacy update manifests, so this will throw in the normal case. What was the behavior before this series of changes when there was no commitTime field in the manifest? We'll need to mirror that behavior.

Reply:
This behavior was changed because `UpdatesUtils.parseDateString()` now takes `String` instead of `String?`, so we have to do the null check here.

Reply:
I've redone this so that the diff from before the state change PR is smaller and easier to see.